firmware/robot2015/src-ctrl/modules/robot-expander-inputs.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include <mbed.h>
+#include <rtos.h>
+
+#include <mcp23017.hpp>
+#include <io-expander.hpp>
+#include <helper-funcs.hpp>
+#include <logger.hpp>
+#include <Console.hpp>
+
+#include "pins-ctrl-2015.hpp"
+#include "task-signals.hpp"
+
+/**
+ * A DigitalInOut class meant to replicate basic functionality of the
+ * mBed DigitalOut and DigitalIn on the io-expander.
+ */
+class RobotExpanderInputs {
+public:
+    /// Other constructors for creating objects for pinouts
+    RobotExpanderInputs(MCP23017* mcp, PinName interruptPin)
+        : interruptThread_(&RobotExpanderInputs::ThreadHelper, this,
+                           osPriorityBelowNormal, 0.33 * DEFAULT_STACK_SIZE),
+          mcp23017_(mcp),
+          interrupt_(interruptPin) {
+        interrupt_.fall(this, &RobotExpanderInputs::ISR);
+        interrupt_.mode(PullUp);
+    }
+
+    uint8_t RotarySelector() {
+        uint8_t reading = 0;
+        const std::array<MCP23017::ExpPinName, 4> kPins = {
+            RJ_HEX_SWITCH_BIT0, RJ_HEX_SWITCH_BIT1, RJ_HEX_SWITCH_BIT2,
+            RJ_HEX_SWITCH_BIT3};
+
+        for (size_t i = 0; i < kPins.size(); i++)
+            reading |= mcp23017_->readPin(kPins[i]) << i;
+        return reading;
+    }
+
+    uint8_t DipSwitches() {
```

Comment:
If you want to, you can factor out most of this method and the RotarySelector method into a "read pin array" or something so that you don't have to write the same code twice.